src/taskgraph/parameters.py

```diff
@@ -79,7 +80,13 @@ def get_version(repo_path):
 
 def _get_defaults(repo_root=None):
     repo_path = repo_root or os.getcwd()
-    repo = get_repository(repo_path)
+    try:
+        repo = get_repository(repo_path)
+    except RuntimeError:
+        # Use fake values if no repo is detected.
+        repo = Mock(branch="", head_rev="", tool="git")
```

Comment:
You could do `repo = Mock(spec_set=GitRepository, branch="", head_rev="", tool="git")` so we can only set values in `GitRepository` here.

Reply:
I just picked `git` because tool needs to be one of `hg` or `git` (and I figure `hg` is being phased out). But this is being used to mock out an hg repo, so I think it's fine to just leave it as a pure mock here and keep it updated if we depend on anything else.